crates/workspace/src/workspace.rs

```diff
@@ -3122,21 +3123,46 @@ impl Workspace {
         }))
     }
 
+    pub fn join_all_panes(&mut self, cx: &mut ViewContext<Self>) {
+        let active_item = self.active_pane.read(cx).active_item().take();
+        for pane in self.panes.clone() {
```

Comment:
```suggestion
        for pane in &self.panes {
```

There's a `pane.clone()` happening below, so no need to clone the entire collection.